Research/lva-node-app/public/js/main.js

```diff
@@ -0,0 +1,656 @@
+/**
+ * @fileoverview
+ * This file contains the majority of the functionality used across the various web pages, in the following order
+ * 1: universal functions
+ * 2: functions related to invoking methods
+ * 3: camera related functions
+ */
+
+
+//immutable global variables are written in all caps
+const PORT = 5000;
+
+/**
+ * @global variables. Mutable global variables in javascript are typically named in camelCase
+ */
+var graphTopologies = {};
+var graphInstances = {};
+var cameras = {};
+
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// universal functions
+
+/**
+* get global variables from server, Returns a promise so that functions that depend on these variables can wait
+* @returns {Promise<any>}
+*/
+function getCamerasFromServer()
+{
+    return new Promise((resolve, reject) =>
+    {
+        let request = prepareRequest(`http://localhost:${PORT}/globals`, "GET");
+        request.onreadystatechange = function ()
+        {
+            if (request.readyState == 4)
+            {
+                if (request.status == 200)
+                {
+                    let response = JSON.parse(request.response);
+                    cameras = response.cameras;
+                    resolve(response);
+                }
+                else
+                {
+                    reject("Get Globals rejection");
+                }
+            }
+            else if (request.status != 200)
+            {
+                reject("Server not responding");
+            }
+        }
+        request.send();
+    });
+}
+
+/**
+* send current values to the server to be stored for later 
+*/
+function sendCamerasToServer()
+{
+    let globals =
+    {
+        cameras: cameras
+    };
+    let request = prepareRequest(`http://localhost:${PORT}/globals`, "PUT");
+    request.send(JSON.stringify(globals));
```

Comment:
this seems like it still needs onreadystatechange and a promise. right now, if anything fails there's no handling. plus send is async, so this will return before the send has completed (or failed) with no way for the user to wait on it.

Reply:
I've added a handler but I think the only time we would ever get a failure is if there was a timeout